### PR TITLE
Add CannonKeys Balance PCB to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding Balance keyboard to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/balance/balance.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "balance.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (clockwise) {
```
Comment (drashna) [inline keyboards/cannonkeys/balance/balance.c:20]:
```suggestion
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (clockwise) {
```

File: keyboards/cannonkeys/balance/config.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0xBA77
+#define DEVICE_VER      0x0001
+#define MANUFACTURER CannonKeys
+#define PRODUCT Balance
+#define DESCRIPTION Balance Keyboard
```
Comment (drashna) [inline keyboards/cannonkeys/balance/config.h:26]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
n

**Review by drashna — APPROVED** [review]

File: keyboards/cannonkeys/balance/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
+CUSTOM_MATRIX = no          # Custom matrix file
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/cannonkeys/balance/rules.mk:23]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks

**Pull request merged**